backend/python/citizen_reports.py

```
@@ -205,6 +222,27 @@ async def submit_citizen_report(
+        cooldown_key = f"citizen_report_cooldown:{identifier}"
+        try:
+            if redis_client.exists(cooldown_key):
+                ttl = redis_client.ttl(cooldown_key)
+                raise HTTPException(
+                    status_code=status.HTTP_429_TOO_MANY_REQUESTS,
+                    detail={
+                        "error": "Submission cooldown",
+                        "message": "Please wait before submitting another report. This helps prevent spam.",
+                        "retry_after": max(1, ttl),
+                    },
+                )
```

Ruff (0.15.0)

[warning] Abstract `raise` to an inner function

(TRY301)